File: app/src/main/java/com/group15/fitnesstracker/db/BodyPartMeasureRecording.java
package com.group15.fitnesstracker.db;

import java.util.Date;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(foreignKeys = @ForeignKey(entity = BodyMeasureRecording.class,parentColumns = "recordingId",childColumns = "recordingId"))
public class BodyPartMeasureRecording {
    @PrimaryKey
    public int recordingId;

    public Date date;

    public String bodypart;

    public double bodypartSize;
}

File: app/src/main/java/com/group15/fitnesstracker/db/BodyPartMeasureRecording.kt
package com.group15.fitnesstracker.db

import java.util.Date

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(foreignKeys = [ForeignKey(entity = BodyMeasureRecording::class, parentColumns = ["recordingId"], childColumns = ["recordingId"]),
                       ForeignKey(entity = User::class, parentColumns = ["id"], childColumns = ["userId"])])
data class BodyPartMeasureRecording (
        @PrimaryKey(autoGenerate = true) var recordingId: Int = 0,
        val userId: Int?,
        var date: Date? = Date(),
        var bodyPart: String?,
        var bodyPartSize: Double?
)